x-pack/plugins/apm/public/components/app/TransactionDetails/Transaction/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -231,43 +236,44 @@ function createGetTransactionById(itemsById: IWaterfallIndex) {
 }
 
 export function getWaterfall(
-  hits: Array<Span | Transaction>,
-  entryTransaction: Transaction
+  hits: TraceAPIResponse['trace'],
+  errorsPerTransaction: TraceAPIResponse['errorsPerTransaction'],
+  entryTransactionId?: Transaction['transaction']['id']
 ): IWaterfall {
-  if (isEmpty(hits)) {
+  if (isEmpty(hits) || !entryTransactionId) {
     return {
       services: [],
       duration: 0,
       items: [],
       itemsById: {},
       getTransactionById: () => undefined,
+      errorsPerTransaction,
       serviceColors: {}
     };
   }
 
-  const filteredHits = hits
-    .filter(hit => {
-      const docType = hit.processor.event;
-      return ['span', 'transaction'].includes(docType);
-    })
-    .map(hit => {
-      const docType = hit.processor.event;
-      switch (docType) {
-        case 'span':
-          return getSpanItem(hit as Span);
-        case 'transaction':
-          return getTransactionItem(hit as Transaction);
-        default:
-          throw new Error(`Unknown type ${docType}`);
-      }
-    });
+  const waterfallItems = hits.map(hit => {
```

Comment:
Thanks for renaming this. I think `waterfallItems` makes more sense here. Unfortunately we also have a method called `getWaterfallItems` that we call `items`. This seems strange.

I know this it not something you added but would be great if you have ideas for improving this.

Reply:
i ended up renaming `items` to `orderedItems` since it's only used to render the timeline in the correct order